README.md

```diff
@@ -269,6 +269,16 @@ A `JsonApiClient::ResultSet` object should be paginatable with both `kaminari` a
 Person.where(name: 'Jeff').all
 ```
 
+## Headers
+
+You can specify the headers to send with all requests within a block with `with_headers`.
```

Comment:
It's not thread safe, but it doesn't modify the single connection object. It adds a hash of custom headers to the resource class and the connection will add those headers for requests of that resource and then remove them when the request is done.

Basically, something like this on the resource:

```ruby
def with_headers(headers)
  @custom_headers = headers
  yield
ensure
  @custom_headers = {}
end
```